script/path.js

```diff
@@ -0,0 +1,361 @@
+							var cName = child.children('.row_key').text();
+							if(cName < lk) {
+								curPrev = child.attr('id');
+							}
+						}
+					});
+					if(curPrev == null) {
+						row.insertAfter(wRow);
+					} else {
+						row.insertAfter(outfit.find('#' + curPrev));
+					}
+				} else {
+					$('div#' + row.attr('id') + ' > div.row_val > span', outfit).text(num);
+					$('div#' + row.attr('id') + ' .tooltip .numAvailable', outfit).text(have - num);
+				}
+				if(num === 0) {
+					$('.dnBtn', row).addClass('disabled');
+					$('.dnManyBtn', row).addClass('disabled');
+				} else {
+					$('.dnBtn', row).removeClass('disabled');
+					$('.dnManyBtn', row).removeClass('disabled');
+				}
+				if(num == have || space < Path.getWeight(k)) {
+					$('.upBtn', row).addClass('disabled');
+					$('.upManyBtn', row).addClass('disabled');
+				} else {
+					$('.upBtn', row).removeClass('disabled');
+					$('.upManyBtn', row).removeClass('disabled');
+				}
+			} else if(have === 0 && row.length > 0) {
+				row.remove();
+			}
+		}
+
+		Path.updateBagSpace(currentBagCapacity);
+
+	},
+
+	updateBagSpace: function(currentBagCapacity) {
+		// Update bagspace
+		$('#bagspace').text(_('free {0}/{1}', Math.floor(Path.getCapacity() - currentBagCapacity) , Path.getCapacity()));
+
+		if(Path.outfit['cured meat'] > 0) {
+			Button.setDisabled($('#embarkButton'), false);
+		} else {
+			Button.setDisabled($('#embarkButton'), true);
+		}
+
+	},
+	
+	createOutfittingRow: function(key, num, store) {
+		if(!store.name) store.name = _(key);
+		var row = $('<div>').attr('id', 'outfit_row_' + key.replace(' ', '-')).addClass('outfitRow').attr('key',key);
+		$('<div>').addClass('row_key').text(store.name).appendTo(row);
+		var val = $('<div>').addClass('row_val').appendTo(row);
+		
+		$('<span>').text(num).appendTo(val);
+		$('<div>').addClass('upBtn').appendTo(val).click([1], Path.increaseSupply);
+		$('<div>').addClass('dnBtn').appendTo(val).click([1], Path.decreaseSupply);
+		$('<div>').addClass('upManyBtn').appendTo(val).click([10], Path.increaseSupply);
+		$('<div>').addClass('dnManyBtn').appendTo(val).click([10], Path.decreaseSupply);
+		$('<div>').addClass('clear').appendTo(row);
+		
+		var numAvailable = $SM.get('stores["'+key+'"]', true);
+		var tt = $('<div>').addClass('tooltip bottom right').appendTo(row);
+
+		if(store.type == 'weapon') {
+			$('<div>').addClass('row_key').text(_('damage')).appendTo(tt);
+			$('<div>').addClass('row_val').text(World.getDamage(key)).appendTo(tt);
+		} else if(store.type == 'tool' && store.desc != "undefined") {
+			$('<div>').addClass('row_key').text(store.desc).appendTo(tt);
+		}
+
+		$('<div>').addClass('row_key').text(_('weight')).appendTo(tt);
+		$('<div>').addClass('row_val').text(Path.getWeight(key)).appendTo(tt);
+		$('<div>').addClass('row_key').text(_('available')).appendTo(tt);
+		$('<div>').addClass('row_val').addClass('numAvailable').text(numAvailable).appendTo(tt);
+		
+		return row;
+	},
```

Comment:
This method is fairly long, nearing the size of a bloater. Much of the jQuery code for adding classes and changing text seems fairly repetitve, so it could probably be made into a helper method and then, using a for loop through classes, much of this code would be simplified.